src/net_processing.cpp

```diff
@@ -2685,8 +2686,12 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
 
         // Potentially mark this peer as a preferred download peer.
         {
-        LOCK(cs_main);
-        UpdatePreferredDownload(pfrom, State(pfrom.GetId()));
+            LOCK(cs_main);
+            CNodeState* state = State(pfrom.GetId());
```

Comment:
It may be better to move this code inline in one commit, separate from changing `nPreferredDownload` from a global to a member variable, so you're not mixing move-only changes and refactor changes in the same commit.

This code could also be cleaned up in a couple of ways now that it's inline (but that doesn't need to happen in this PR):

- `m_num_preferred_download_peers -= state->fPreferredDownload;` can be removed, since `version` processing can only happen once for each peer, and `fPreferredDownload` is initialized to false, so this line is always a no-op.
- `// Whether this node should be marked as a preferred download node.` is redundant with the `// Potentially mark this peer as a preferred download peer.` comment above and can be removed.

Reply:
Good idea, I split the inlining and clean ups into a separate commit.